ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java

```diff
@@ -239,6 +240,13 @@ public RestMethod getRestMethod() {
   }
 
   @Override
+  public Object setArg(String key, Object value) {
+    Object oldVal = allArgs.put(key, value);
+    allArgsReadOnly = Collections.unmodifiableMap(allArgs);
```

Comment:
why is this step needed? changing the underlying map should have changed this too.